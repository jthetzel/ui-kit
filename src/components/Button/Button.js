import React from 'react';
import styled from 'react-emotion';

const Button = styled.button`
  background-color: salmon;
  border: 1px solid indianred;
  color: snow;
`;

/**
 * @component
 */
export default props => <Button {...props} />;
