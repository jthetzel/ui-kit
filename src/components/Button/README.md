Button

```js
<Button>Push Me</Button>
```