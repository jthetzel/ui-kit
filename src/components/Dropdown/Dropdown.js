import React from 'react';
import PropTypes from 'prop-types';

import ClickOutside from '../ClickOutside';
import withToggle from '../withToggle';
import Container from './Container';

const Dropdown = ({
  renderHandler, renderContent, toggledOn, toggle, show, hide
}) => (
  <ClickOutside onClickOutside={hide}>
    <Container>
      {renderHandler({
        hide,
        show,
        toggle,
        toggledOn
      })}
      {toggledOn &&
        renderContent({
          hide,
          show,
          toggle,
          toggledOn
        })}
    </Container>
  </ClickOutside>
);

Dropdown.propTypes = {
  hide: PropTypes.func.isRequired,
  renderContent: PropTypes.func.isRequired,
  renderHandler: PropTypes.func.isRequired,
  show: PropTypes.func.isRequired,
  toggle: PropTypes.func.isRequired,
  toggledOn: PropTypes.bool.isRequired
};

export default withToggle(Dropdown);
