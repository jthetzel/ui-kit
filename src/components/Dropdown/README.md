Simple Dropdown

```js
const Handler = ({ toggle, toggledOn }) => (
  <button onClick={toggle}>{toggledOn ? "close" : "open"}</button>
);

const Content = () => <h2>Content</h2>;

<Dropdown renderHandler={Handler} renderContent={Content} />;
```
