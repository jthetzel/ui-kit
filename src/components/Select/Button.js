import styled from 'react-emotion';

export default styled.div`
  position: relative;
  border-top: 1px solid black;
  border-bottom: 1px solid black;
  padding: 20px 0;
  user-select: none;
  width: 100%;
  
  &:hover {
    cursor: pointer;
  }
`;
