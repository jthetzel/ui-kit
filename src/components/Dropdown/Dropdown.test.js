/* eslint function-paren-newline: 0 */

import React from 'react';
import { shallow } from 'enzyme';
// import simulant from 'simulant';
import toJson from 'enzyme-to-json';
import Dropdown from '../Dropdown';
import Button from '../Button/Button';

test('Dropdown does not crash', () => {
  const tree = shallow(
    <Dropdown
      renderHandler={({ toggle }) => <Button onClick={toggle}>Button</Button>}
      renderContent={() => 'Children'}
    />
  );

  expect(toJson(tree)).toMatchSnapshot();
});

// test('Dropdown test setChildNodeRef', () => {
//   mount(
//     <Dropdown
//       renderHandler={({ toggle }) => <Button onClick={toggle}>Button</Button>}
//       renderContent={() => 'Children'}
//     />
//   );
// });

// test('Dropdown simulate renderHandler onClick', () => {
//   const wrapper = shallow(
//     <Dropdown
//       renderHandler={toggle => <Button onClick={toggle}>Button</Button>}
//       renderContent={() => 'Children'}
//     />
//   );

//   const btn = wrapper.find(Button).first();
//   expect(wrapper.state().isOpen).toEqual(false);
//   btn.simulate('click');
//   expect(wrapper.state().isOpen).toEqual(true);
// });

// test('click outside of component', () => {
//   const wrapper = shallow(
//     <Dropdown
//       renderHandler={toggle => <Button onClick={toggle}>Button</Button>}
//       renderContent={() => 'Children'}
//     />
//   );
//   const btn = wrapper.find(Button).first();
//   expect(wrapper.state().isOpen).toEqual(false);

//   btn.simulate('click');
//   expect(wrapper.state().isOpen).toEqual(true);

//   simulant.fire(document, 'click');
//   expect(wrapper.state().isOpen).toEqual(false);
// });
