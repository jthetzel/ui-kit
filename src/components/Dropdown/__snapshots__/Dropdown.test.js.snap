// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown does not crash 1`] = `
<withHandlers(Dropdown)
  renderContent={[Function]}
  renderHandler={[Function]}
  toggle={[Function]}
  toggledOn={false}
/>
`;
